lib/logstash/inputs/beats.rb

```diff
@@ -152,118 +164,28 @@ def target_codec_on_field?
   end
 
   def run(output_queue)
-    @output_queue = output_queue
-    start_buffer_broker
-
-    while !stop? do
-      # Wrapping the accept call into a CircuitBreaker
-      if @circuit_breaker.closed?
-        connection = @lumberjack.accept # call that creates a new connection
-        # if the connection is nil the connection was closed upstream,
-        # so we will try in another iteration to recover or stop.
-        next if connection.nil?
-
-        Thread.new do
-          handle_new_connection(connection)
-        end
-      else
-        @logger.warn("Beats input: the pipeline is blocked, temporary refusing new connection.",
-                     :reconnect_backoff_sleep => RECONNECT_BACKOFF_SLEEP)
-        sleep(RECONNECT_BACKOFF_SLEEP)
-      end
-    end
+    message_listener = MessageListener.new(output_queue, self)
```

Comment:
I can't remove the `self` for now because we need to have access to the plugin to be able to decorate the event with the correct tags or types.

```
    def transform(event)
      copy_beat_hostname(event)
      decorate(event)
      event
    end
```